aws_lambda_powertools/utilities/idempotency/serialization/base.py

```diff
@@ -0,0 +1,23 @@
+"""
+Serialization for supporting idempotency
+"""
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+logger = logging.getLogger(__name__)
+
+
+class BaseDictSerializer(ABC):
```

Comment:
What do you think calling this `BaseSerializer` instead? It's more generic.

Reply:
I also thought about it.
This comes back to the question if there will be a refactor to 'direct' serialization (see  https://github.com/aws-powertools/powertools-lambda-python/pull/2951#issuecomment-1693242636)
Because if we would like a refactor, and you already use *BaseSerializer*, you cannot addd an intermediate class of * BaseDictSerializer* and only change the internal implementation

Reply:
updated to BaseIdempotencySerializer